Improves type annotations

chuck-flowers opened the pull request:
Updates type annotations in `Logbook` and `Section`

Adds and updates type annotations to reflect the actual implementations of the
modules/classes.

kristijanhusak commented on lua/orgmode/parser/section.lua:
```diff
@@ -34,8 +34,37 @@ local config = require('orgmode.config')
+---@field tags string[]
+---@field title string
+---@field todo_keyword_node unknown
+local NewSectionOptions = {}
```
There's actually no need to define an empty variable to get type annotations. LSP should know to pick it up even without it.

chuck-flowers replied:
I've removed the two unused variables that I added as part of this contribution

kristijanhusak reviewed (changes requested):
Thanks for the PR. Just address that one comment and it's good to go.

Just a note if you are maybe trying to use the internals for a separate plugin, backward compatibility is not guaranteed for the internals. If you think you need some of these things exposed, I would suggest to open up a separate PR and introduce a public api through `orgmode.api` files.

Edit: `node` can be typed as `userdata` instead of `unknown` and `any`.

chuck-flowers commented:
I added these type annotations as part of some fixes I'm contributing to this plugin. It took me a minute to figure out what types certain variables should be from the call site, so once I figured it out by looking at `Section` and `Logbook` I updated the type annotations to save myself or the next person some time.

kristijanhusak commented on lua/orgmode/parser/logbook.lua:
```diff
@@ -134,7 +134,8 @@ function Logbook:recalculate_estimate(line)
 
 ---@param lines string
----@param dates string
+---@param node unknown
```
You can type this as `userdata`, since tree-sitter nodes are of that type. Same thing for the `_parse_clocks` params.

chuck-flowers replied:
I've update this as well

kristijanhusak commented:
Thanks!
